### ISSUE TITLE
Specifying targetArn when Publish SNS message seems to no longer work

### ISSUE THREAD

**Issue by maccamlcQ** [localstack/localstack#1933]
Using 0.10.7.

Previously with 0.10.4 I had some tests that were publishing a SNS message, with the topic subscribed to SQS queue.

The PublishRequest was specified by targetArn. In 0.10.4 this worked.

However since upgrading to 0.10.7 these tests broke. I discovered that changing to specify topicArn instead, they succeed again.

Example.

This worked in 0.10.4 but broke in 0.10.7

```
        amazonSNSAsync.publish(new PublishRequest()
                .withTargetArn(topicArn)
                .withSubject("Listener 1 Subject")
                .withMessage("Listener 1 Message")
        );
```

Changing 

```
        amazonSNSAsync.publish(new PublishRequest()
                .withTopicArn(topicArn)
                .withSubject("Listener 1 Subject")
                .withMessage("Listener 1 Message")
        );
```

makes it work successfully.

I believe that targetArn or topicArn should both still be possible to use?

**Comment by whummer** [comment]
Thanks for reporting @maccamlcQ . Is this a potential duplicate of #1876?

**Comment by maccamlcQ** [comment]
Thanks @whummer . Yes, it seems to be a duplicate. Apologies I didn't find that first

**Comment by whummer** [comment]
This should be fixed in #2270 - can you please give it a try with the latest version? Please report here if the problem persists. Thanks
